psychopy/event.py

```diff
@@ -371,27 +377,33 @@ def waitKeys(maxWait=float('inf'), keyList=None, modifiers=False,
             reset.
         clearBuffer : **True** or False
             Whether to clear the keyboard buffer (and discard preceding
-            keypresses) or not.
+            keypresses) before starting to monitor for keypresses. If
+            `keyList` is specified, only those keys are removed from the
+            keyboard buffer.
 
     Returns None if times out.
 
     """
-    # NB pygame.event does have a wait() function that will
-    # do this and maybe leave more cpu idle time?
-    key = None
     if clearBuffer:  # Only consider keypresses from here onwards.
-        clearEvents('keyboard')
+        if not keyList:
+            clearEvents('keyboard')
+        else:
+            getKeys(keyList=keyList)
```

Comment:
@lindeloev, do you think this is a viable solution? Was the easiest approach I could think of!